### PR TITLE
Incorporated changes from PR #38 on fulcro incubator

### PR DESCRIPTION
**Pull request by abhi18av**
Porting over the patch from https://github.com/fulcro-legacy/fulcro-incubator/commit/38659c19cc8caa20167d4649242039d7a35dfae1 to fulcro3 repo

### CODE REVIEW

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc [hunk 1]
```diff
@@ -314,18 +314,38 @@
...
+;;                             (swap! to-signal conj c)))
+;;;;;;;;
+;; FIXME: Resolve references to prim namespace => /portfolio/fulcro2/src/main/fulcro/client/primitives.cljc
+                              (let [mounted-target-class (reduce (fn [acc {:keys [dispatch-key component]}]
+                                                                   (when (= ::current-route dispatch-key)
+                                                                     (reduced component)))
+                                                           nil
+                                                           (some-> component (prim/get-query state-map)
+                                                             prim/query->ast :children))
+                                    mounted-targets      (prim/class->all reconciler mounted-target-class)]
```
Comment (abhi18av) [inline src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: Porting `prim/class->all` to it's `comp/class->all` counterpart.

**Comment by abhi18av** [comment]
@awkay, I've have ported over the bug-fix now and let me know if there's anything you'd like me to alter in this PR.

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc [hunk 2]
```diff
@@ -346,6 +366,9 @@
   ([this-or-app relative-class-or-instance new-route]
    (change-route-relative this-or-app relative-class-or-instance new-route {}))
   ([app-or-comp relative-class-or-instance new-route timeouts]
+ ;; TODO: Add next lines
+ ;;   (when-not (seq (proposed-new-path this-or-reconciler relative-class-or-instance new-route))
```
Comment (abhi18av) [inline src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: Just ported the `this-or-reconciler` to newer `this-or-app`.

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc [hunk 3]
```diff
@@ -314,18 +314,38 @@
...
                      (when (and component (router? component))
                        (let [new-target    (first new-path-remaining)
                              router-ident  (comp/get-ident component {})
                              active-target (get-in state-map (conj router-ident ::current-route))
                              {:keys [target]} (get-in state-map (conj router-ident ::pending-route))
-                             next-router   (some #(ast-node-for-live-router app %) (:children node))]
+;; TODO: Remove next line
+;;                             next-router   (some #(ast-node-for-live-router app %) (:children node))]
+;; FIXME: Resolve reference to reconciler => /portfolio/fulcro2/src/main/fulcro/client/primitives.cljc
+                               next-router   (some #(ast-node-for-live-router reconciler %) children)]
```
[the remark this answers is not in the document]
Comment (awkay) [reply src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: yep

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc [hunk 4]
```diff
@@ -314,18 +314,38 @@
...
-        _          (loop [{:keys [component] :as node} root new-path-remaining new-path]
+;; TODO: Remove next line
+;;        _          (loop [{:keys [component] :as node} root new-path-remaining new-path]
+        _          (loop [{:keys [component children] :as node} root new-path-remaining new-path]
                      (when (and component (router? component))
                        (let [new-target    (first new-path-remaining)
                              router-ident  (comp/get-ident component {})
                              active-target (get-in state-map (conj router-ident ::current-route))
                              {:keys [target]} (get-in state-map (conj router-ident ::pending-route))
-                             next-router   (some #(ast-node-for-live-router app %) (:children node))]
```
Comment (awkay) [inline src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: You can see here that app is exactly how I ported the old version

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc [hunk 5]
```diff
@@ -314,18 +314,38 @@
...
+;; TODO: Remove next lines
+;;                           (when-let [c (comp/ident->any app active-target)]
+;;                             (swap! to-signal conj c)))
+;;;;;;;;
+;; FIXME: Resolve references to prim namespace => /portfolio/fulcro2/src/main/fulcro/client/primitives.cljc
+                              (let [mounted-target-class (reduce (fn [acc {:keys [dispatch-key component]}]
+                                                                   (when (= ::current-route dispatch-key)
+                                                                     (reduced component)))
+                                                           nil
+                                                           (some-> component (prim/get-query state-map)
```
[the remark this answers is not in the document]
Comment (awkay) [reply src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: yes

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc [hunk 6]
```diff
@@ -314,18 +314,38 @@
...
+                                                             prim/query->ast :children))
```
[the remark this answers is not in the document]
Comment (awkay) [reply src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: Yes, EQL is where those live now.

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc (hunk 1 above)
Comment (awkay) [reply src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: yes

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc (hunk 2 above)
Comment (awkay) [reply src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: If you look at the other arity of the function I chose the name app-or-comp, so keep consistent

File: src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc (hunk 4 above)
Comment (abhi18av) [reply src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc]: Yup, noted :+1:

**Comment by abhi18av** [comment]
Thanks @awkay 😊

**Pull request merged**